node/metrics.go

```diff
@@ -150,3 +169,59 @@ func (g *Metrics) AcceptBatches(status string, batchSize uint64) {
 	g.AccuBatches.WithLabelValues("number", status).Inc()
 	g.AccuBatches.WithLabelValues("size", status).Add(float64(batchSize))
 }
+
+func (g *Metrics) collectOnchainMetrics() {
+	ticker := time.NewTicker(time.Duration(uint64(g.onchainMetricsInterval)))
+	defer ticker.Stop()
+
+	// 3 chain RPC calls in each cycle.
+	for range ticker.C {
+		ctx := context.Background()
+		blockNum, err := g.tx.GetCurrentBlockNumber(ctx)
+		if err != nil {
+			g.logger.Error("Failed to query chain RPC for current block number", "err", err)
+			continue
+		}
+		bitmaps, err := g.tx.GetQuorumBitmapForOperatorsAtBlockNumber(ctx, []core.OperatorID{g.operatorId}, blockNum)
+		if err != nil {
+			g.logger.Error("Failed to query chain RPC for quorum bitmap", "blockNumber", blockNum, "err", err)
+			continue
+		}
+		quorumIds := eth.BitmapToQuorumIds(bitmaps[0])
+		if len(quorumIds) == 0 {
+			g.logger.Warn("This node is currently not in any quorum", "blockNumber", blockNum, "operatorId", g.operatorId.Hex())
+			continue
+		}
+		state, err := g.chainState.GetOperatorState(ctx, uint(blockNum), quorumIds)
+		if err != nil {
+			g.logger.Error("Failed to query chain RPC for operator state", "blockNumber", blockNum, "quorumIds", quorumIds, "err", err)
+			continue
+		}
+		type OperatorStakeShare struct {
+			operatorId core.OperatorID
+			stakeShare float64
+		}
+		for q, operators := range state.Operators {
+			operatorStakeShares := make([]*OperatorStakeShare, 0)
+			for opId, opInfo := range operators {
+				share, _ := new(big.Int).Div(new(big.Int).Mul(opInfo.Stake, big.NewInt(10000)), state.Totals[q].Stake).Float64()
+				operatorStakeShares = append(operatorStakeShares, &OperatorStakeShare{operatorId: opId, stakeShare: share})
+			}
+			// Descending order by stake share in the quorum.
+			sort.Slice(operatorStakeShares, func(i, j int) bool {
+				if operatorStakeShares[i].stakeShare == operatorStakeShares[j].stakeShare {
+					return operatorStakeShares[i].operatorId.Hex() < operatorStakeShares[j].operatorId.Hex()
+				}
+				return operatorStakeShares[i].stakeShare > operatorStakeShares[j].stakeShare
+			})
+			for i, op := range operatorStakeShares {
+				if op.operatorId == g.operatorId {
+					g.Registered.WithLabelValues(string(q), "stake_share").Set(op.stakeShare)
```

Comment:
what is the point of sending 2 different metrics? instead of sending 1 with both label (metric will also need to be updated the two labels)

Reply:
We need both stake share and the rank of the operator's share across operators.